experimental/framework/fluid-llm/src/shared-tree-diff/sharedTreeBranchManager.ts

```diff
@@ -0,0 +1,261 @@
+import { type TreeArrayNode } from "@fluidframework/tree";
+import type * as z from "zod";
+
+import {
+	sharedTreeDiff,
+	type Difference,
+	type DifferenceChange,
+	type DifferenceCreate,
+	type DifferenceMove,
+	type DifferenceRemove,
+	type ObjectPath,
+} from "./sharedTreeDiff.js";
+import { isTreeMapNode, isTreeArrayNode, sharedTreeTraverse } from "./utils.js";
+
+/**
+ * Manages determining the differences between two branches of a SharedTree represented as an actual tree node or a plain javascript object
+ * and applies said differences to the original SharedTree branch.
+ */
+export class SharedTreeBranchManager {
+	private readonly objectSchema?: z.Schema;
```

Comment:
Nit: Since this is for any kind of node schema, should this be `treeSchema` or `nodeSchema`?